src/engine/errors.rs

```diff
@@ -0,0 +1,60 @@
+use std::{error::Error, fmt::Display};
+
+use tonic::codegen::http::StatusCode;
+
+#[derive(Debug)]
+pub struct EngineError {
+    error: anyhow::Error,
+    pub code: StatusCode,
+}
+
+impl Error for EngineError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        self.error.source()
+    }
+
+    #[allow(deprecated)]
+    fn description(&self) -> &str {
+        self.error.description()
+    }
```

Comment:
Is this really required if you implement the `Display` trait? (See [here](https://doc.rust-lang.org/std/error/trait.Error.html#method.description))

Reply:
It is only deprecated, if another piece of code/library uses the deprecated call, the old behavior still stands until removal